services/wallet/router_send_type.go

```diff
@@ -0,0 +1,169 @@
+package wallet
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/services/wallet/bigint"
+	"github.com/status-im/status-go/services/wallet/bridge"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	"github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/transactions"
+)
+
+type SendType int
+
+const (
+	Transfer SendType = iota
+	ENSRegister
+	ENSRelease
+	ENSSetPubKey
+	StickersBuy
+	Bridge
+	ERC721Transfer
+	ERC1155Transfer
+	Swap
+)
+
+func (s SendType) IsCollectiblesTransfer() bool {
+	return s == ERC721Transfer || s == ERC1155Transfer
+}
+
+func (s SendType) FetchPrices(service *Service, tokenID string) (map[string]float64, error) {
+	symbols := []string{tokenID, "ETH"}
+	if s.IsCollectiblesTransfer() {
+		symbols = []string{"ETH"}
+	}
+
+	pricesMap, err := service.marketManager.FetchPrices(symbols, []string{"USD"})
+	if err != nil {
+		return nil, err
+	}
+	prices := make(map[string]float64, 0)
+	for symbol, pricePerCurrency := range pricesMap {
+		prices[symbol] = pricePerCurrency["USD"]
+	}
+	if s.IsCollectiblesTransfer() {
+		prices[tokenID] = 0
+	}
+	return prices, nil
+}
+
+func (s SendType) FindToken(service *Service, account common.Address, network *params.Network, tokenID string) *token.Token {
+	if !s.IsCollectiblesTransfer() {
+		return service.tokenManager.FindToken(network, tokenID)
+	}
+
+	parts := strings.Split(tokenID, ":")
+	contractAddress := common.HexToAddress(parts[0])
+	collectibleTokenID, success := new(big.Int).SetString(parts[1], 10)
+	if !success {
+		return nil
+	}
+	uniqueID, err := service.collectibles.GetOwnedCollectible(walletCommon.ChainID(network.ChainID), account, contractAddress, collectibleTokenID)
+	if err != nil || uniqueID == nil {
+		return nil
+	}
+
+	return &token.Token{
+		Address:  contractAddress,
+		Symbol:   collectibleTokenID.String(),
+		Decimals: 0,
+		ChainID:  network.ChainID,
+	}
+}
+
+func (s SendType) isTransfer() bool {
+	return s == Transfer || s == Swap || s.IsCollectiblesTransfer()
+}
+
+func (s SendType) needL1Fee() bool {
+	return s != ENSRegister && s != ENSRelease && s != ENSSetPubKey && s != StickersBuy
+}
+
+func (s SendType) canUseBridge(b bridge.Bridge) bool {
+	if s == ERC721Transfer && b.Name() != ERC721TransferString {
+		return false
+	}
+
+	if s != ERC721Transfer && b.Name() == ERC721TransferString {
+		return false
+	}
+
+	if s == ERC1155Transfer && b.Name() != ERC1155TransferString {
+		return false
+	}
+
+	if s != ERC1155Transfer && b.Name() == ERC1155TransferString {
+		return false
+	}
+
+	return true
+}
```

Comment:
Could I make a couple of suggestions for this function:

1. Compatibility map approach:

Map each `SendType` to the expected bridge name. This way, the function looks up the required bridge name for a given `SendType` and compares it directly with the bridge’s name, this reduces the number of conditional statements.

```suggestion
func (s SendType) canUseBridge(b bridge.Bridge) bool {
    compatibilityMap := map[SendType]string{
        ERC721Transfer:  ERC721TransferString,
        ERC1155Transfer: ERC1155TransferString,
    }

    expectedBridge, ok := compatibilityMap[s]
    return !ok || b.Name() == expectedBridge
}
```

2. Switch approach

Directly map `SendType` to the condition without repeating the negative checks. 

```suggestion
func (s SendType) canUseBridge(b bridge.Bridge) bool {
    bridgeName := b.Name()
    switch s {
    case ERC721Transfer:
        return bridgeName == ERC721TransferString
    case ERC1155Transfer:
        return bridgeName == ERC1155TransferString
    default:
        return true // Assumes other types can use any bridge
    }
}
```

Both approaches have their advantages.